hashintel/hash
# Edit Bar on Entity Type Editor shouldn't scroll entire page when animating

## 🌟 What is the purpose of this PR?




When the edit bar on the entity type editor animates in/out, it scrolls the page to keep the content underneath it static. When this functionality was introduced, there was no sidebar on the type editor. Since this was introduced, it's been discovered the method it uses to scroll the page results in a gap appearing underneath the sidebar. 

This PR fixes that by scrolling the right element + makes this functionality work in Firefox, which it never did. 

## 🔗 Related links





- [Asana task](https://app.asana.com/0/1203179076056209/1203462405114281/f) _(internal)_

## 🚫 Blocked by




N/A

## 🔍 What does this change?




- New `EditBarScroller` component which wraps the page with a DOM node it then provides to its children using context, as well as providing access to the "scrolling element". This is the node which will be scrolled. 
- This component is now used by the `LayoutWithSidebar` component
- Updated `useFreezeScrollWhileTransitioning` to read the nodes from this context instead of using `document.documentElement`. 

## 📜 Does this require a change to the docs?



N/A

## ⚠️ Known issues



N/A

## 🐾 Next steps



N/A

## 🛡 What tests cover this?



N/A

## ❓ How to test this?



1.  Make a change to a property in the entity type editor, and be sure the page doesn't shift as the edit bar animates in 
2. Make sure no gap appears beneath the sidebar while this happens
3. Ensure this also works in firefox as expected
